torchtune/models/flamingo/_component_builders.py

```diff
@@ -264,3 +238,507 @@ def flamingo_decoder(
+            Default: False
+        apply_lora_to_output (bool): whether to apply LoRA to the model's final output projection.
+            Default: False
+        patch_size (int): The size of each patch. Used to divide the tiles into patches.
+            E.g. for ``patch_size=40``, a tile of shape (400, 400) will have 10x10 grid of patches
+            with shape (40, 40) each.
+        num_heads (int): The number of attention heads in each transformer layer.
+        clip_embed_dim (int): The dimensionality of each patch embedding in CLIP.
+        clip_num_layers (int): The number of transformer layers.
+        clip_hidden_states (Optional[List[int]]): The indices of CLIP hidden layers to return
+            to return to the encoder projection head. It will return the intermediate results 
+            of the vision transformer layers which will be concatenated with the CLIP output
+            and input into the projection head. For example, ``clip_hidden_states=[0,3]`` will
+            return the embeddings before they go through the first and fourth layers.
+        num_layers_projection (int): The number of transformer layers in the projection head.
+        decoder_embed_dim (int): The dimensionality of the final output embeddings for the decoder.
+        tile_size (int): The size of your image tiles, if the image was tile-cropped in advance. Otherwise,
+            the size of the input image. In this case, the function will consider your image as a single tile.
+        max_num_tiles (int): The maximum number of tiles that can be processed. This is used to
+            determine the size of the positional embeddings.
+        in_channels (int): The number of image input channels.
+        lora_rank (int): rank of each low-rank approximation
+        lora_alpha (float): scaling factor for the low-rank approximation
+        lora_dropout (float): LoRA dropout probability. Default: 0.0
+        use_dora (bool): Whether to use DoRA layers instead of LoRA layers. Default is ``False``.
+        quantize_base: (bool): Whether to quantize base model weights or not. Only applied to base
+            weights within linear layers LoRA is applied to. The final output linear projection is not
+            supported for quantization currently.
+        
+
+    Returns:
+        FlamingoEncoder: Instantiation of Flamingo encoder.
+    """
+    lora_options = {
+        "lora_modules": lora_attn_modules,
+        "apply_lora_to_mlp": apply_lora_to_mlp,
+        "apply_lora_to_output": apply_lora_to_output,
+        "lora_rank": lora_rank,
+        "lora_alpha": lora_alpha,
+        "lora_dropout": lora_dropout,
+        "use_dora": use_dora,
+        "quantize_base": quantize_base,
+    }
+
+    # clip encoder
+    clip_options = {
+        "tile_size": tile_size,
+        "patch_size": patch_size,
+        "embed_dim": clip_embed_dim,
+        "num_layers": clip_num_layers,
+        "num_heads": num_heads,
+        "activation": nn.GELU,
```

Comment:
Maybe it's deliberate but seems like this doesn't match the default in the CLIP builder

Reply:
Deliberate